### PR TITLE
Optimize commission and (re)subscription flows

### PR DESCRIPTION
**Pull request by marcelveldt**
A bunch of (small) optimizations in the commissioning and (re)subscription flows again based on some investigation with various devices. 

- Remove fetching certificates before commissioning (as it only slows down and doesn't do anything)
- Ignore failed commissioned nodes at startup
- Cancel subscription at node removal
- Speed up resolve
- Split up node attribute subscriptions into 2 subscriptions
- Lower the ceiling once again
- Adjust logging of PersistentStorage module of sdk because it was too chatty

### CODE REVIEW

File: matter_server/server/__main__.py [hunk 1]
```diff
@@ -72,6 +72,7 @@ def main() -> None:
     coloredlogs.install(level=args.log_level.upper())
     if not logging.getLogger().isEnabledFor(logging.DEBUG):
         logging.getLogger("chip").setLevel(logging.WARNING)
+        logging.getLogger("PersistentStorage").setLevel(logging.WARNING)
```
Comment (agners) [inline matter_server/server/__main__.py:75]: This is a rather unrelated change and should be a separate commit (that also has the advantage that it appears in changelogs etc.)

File: matter_server/server/device_controller.py [hunk 2]
```diff
@@ -99,12 +109,16 @@ async def initialize(self) -> None:
...
+        nodes: dict[str, dict | None] = self.server.storage.get(DATA_KEY_NODES, {})
         for node_id_str, node_dict in nodes.items():
             node_id = int(node_id_str)
-            # invalidate node data if schema mismatch,
-            # the node will automatically be scheduled for re-interview
-            if node_dict and node_dict.get("interview_version") != SCHEMA_VERSION:
+            if node_dict is None:
+                # ignore non-initialized (left-over) nodes
+                # from failed commissioning attempts
+                continue
```
Comment (agners) [inline matter_server/server/device_controller.py:118]: IMHO we should still retry to interview those. It could be that the user commissioned them, then interviewing failed, he restarts the add-on and now maybe this second time interviewing works... :man_shrugging: 

I understand that now people have probably such stale entries, and this code makes it retry every time. And since people don't see those in Core, they also cant' delete them.

So, as a work around, I think this is fine.

But once we should them, have them cleaned-up, we should no longer ignore them.

File: matter_server/server/device_controller.py [hunk 3]
```diff
@@ -630,59 +655,83 @@ async def _subscribe_node(self, node_id: int) -> None:
...
+        """Setup a single Node AttributePath(s) subscription."""
+        node_id = node.node_id
+        node_logger = LOGGER.getChild(f"[node {node_id}]")
+        assert self.chip_controller is not None
+        node_logger.debug("Setting up attributes and events subscription.")
+        self._last_subscription_attempt[node_id] = 0
+        self.chip_controller.CheckIsActive()
+        assert self.server.loop is not None
+        future = self.server.loop.create_future()
+        device = self.chip_controller.GetConnectedDeviceSync(node_id)
```
Comment (agners) [inline matter_server/server/device_controller.py:717]: Do we need the resolve lock here (since in the comment in `_resolve_node()` it says simultaneous resolving causes SDK crashes).

File: matter_server/server/device_controller.py (hunk 2 above)
Comment (marcelveldt) [reply matter_server/server/device_controller.py:118]: No, we agreed on this. This is old stuff. It can not happen anymore but we need to clean out the old traces.

File: matter_server/server/device_controller.py (hunk 3 above)
Comment (marcelveldt) [reply matter_server/server/device_controller.py:717]: Oof, yes, good catch!

File: matter_server/server/__main__.py (hunk 1 above)
Comment (marcelveldt) [reply matter_server/server/__main__.py:75]: Yes it should be a seperated commit indeed, I'll put it in a seperate commit

Comment (marcelveldt) [reply matter_server/server/__main__.py:75]: but well in the end this will be a squash commit so it doesnt matter after all. I'll add it to the PR desciption

File: matter_server/server/device_controller.py (hunk 3 above)
Comment (marcelveldt) [reply matter_server/server/device_controller.py]: adjusted

File: matter_server/server/device_controller.py [hunk 4]
```diff
@@ -630,59 +658,82 @@ async def _subscribe_node(self, node_id: int) -> None:
...
+        interval_ceiling: int = 60,
+        event_subscriptions: list[Attribute.EventPath] | None = None,
+    ) -> Attribute.SubscriptionTransaction:
+        """Handle Setup of a single Node AttributePath(s) subscription."""
+        node_id = node.node_id
+        node_logger = LOGGER.getChild(f"[node {node_id}]")
+        assert self.chip_controller is not None
+        node_logger.debug("Setting up attributes and events subscription.")
+        self._last_subscription_attempt[node_id] = 0
+        assert self.server.loop is not None
```
Comment (MartinHjelmare) [inline matter_server/server/device_controller.py:717]: We're using the loop attribute multiple times. I'd store it in a local variable first.

**Review by MartinHjelmare — APPROVED** [review]

File: matter_server/server/device_controller.py [hunk 5]
```diff
@@ -131,8 +131,21 @@ async def start(self) -> None:
...
+                # if the node was not available last time we will delay
+                # the first attempt to initialize so that we prioritize nodes
+                # that are probably available so they are back online as soon as
+                # possible and we're not stuck trying to initialize nodes that are offline
+                assert self.server.loop
+                self.server.loop.call_later(
+                    5,
+                    asyncio.create_task,
+                    self._check_interview_and_subscription(node_id),
+                )
```
Comment (MartinHjelmare) [inline matter_server/server/device_controller.py:144]: It's better to have a callback that creates the task so that we don't have a non awaited coroutine possibly lingering if the server is shutdown before the delay.
